drafts/build_a_dao_on_fuel/draft.md

```diff
@@ -0,0 +1,1459 @@
+
+**Sway**: [Sway](https://fuellabs.github.io/sway/) is a domain-specific language used for writing smart contracts on the fuel blockchain.
+
+**ABI**: These are called [Application Binary Interfaces](https://docs.fuel.network/guides/intro-to-sway/contract-abi/); they are the blueprint or interaction models of your smart contract, it is a layout of the detailed functionalities in your smart contract, and how to interact with it.
+
+## Setting up your Environment
+
+In the next few steps, you will set up your development environment to begin building your smart contract and the front end of your Dao.
+
+1. Go to your terminal and run the command `wsl` to start your `wsl` terminal
+
+2. Next, run the command to generate your new contract directory.
+
+```bash
+mkdir Fuel_Dao
+cd Fuel_Dao
+forc new contract
+```
+
+## Writing your Smart Contract
+
+Head to the `main.sw` contract file, where you will implement your main contracts. Following the next few steps, you will replace the code therein.
+
+### 1. Importing from the modules 
+
+First, you will import the following modules from the sway standard library; 
+  
+- `asset::transfer`: This provides the functionality for transferring assets.
+
+- `auth::msg_sender` - The module retrieves the address of the sender of the contract or a transaction.
+  
+- `block::height` - The module provides information about the current block height or block number
+  
+- `call_frames::msg_asset_id`- This provides information about the asset (token) associated with the current sent transaction.
+  
+- `context` - This module provides context-related functionality related to the execution context of your contract.
+  
+`hash::Hash` - This provides hashing-related functionality, such as integrity verification, cryptographic operations, and generating unique identifiers.
+
+  Copy and add the code below to your `main.sw` contract.
+  
+
+  ```rust
+    use std::{
+      asset::transfer,
+      auth::msg_sender,
+      block::height,
+      call_frames::msg_asset_id,
+      context::{
+        msg_amount,
+        this_balance,
+      },
+      hash::Hash,
+    };
+  ```
+
+
+### 2. Importing Other Library Contracts
+Next, you define the file as a contract file and import other contract libraries and interfaces from the following `crates`.
+
+- `error` - define the error handlers for your contract.
+- `data_structre` - defines functionalities for managing proposals and voting and defines methods for initializing your DAO.
+- `events` - defines a library of multiple `structs` to **emit events** when certain instances or functions are executed on-chain.
+- `utils` - These will be instances that are emitted on-chain when certain events are met.
+- `interface` - this is where you will implement your contract ABIs
+
+All of these are libraries you will be creating later in this tutorial.
+
+  Copy and replace the existing code in your `main.sw` file with the one below.
```

Comment:
Please remove the leading space before "Copy and replace the existing code in your `main.sw` file with the one below."

Reply:
Thanks for Apotting this out 
-  Corrected 
> Copy and replace the existing code in your `main.sw` file with the one below.